### DISCUSSION TITLE
Would using a combat macro instead of CSS speed up script execution?

### DISCUSSION THREAD

**Discussion by digitalvint** [loathers/garbage-collector#1841] [Ideas]
Making the combat server side instead of client side, would it reduce script running time? When i watch garbo run, it seem to hang for about a second for each combat which I assume is the combat doing its thing.

Could even be a choice whether to use CSS or combat macro.

**Comment by gausie** [comment]
There is too much code conditional on things that macros don't know about. Our combat system is smart and will try to do as much as possible via macros, but the reality is that most of it can't be. You could try running `garbo quick` for faster combats if you'd like.
